### ISSUE TITLE
Parameter arguments in function get_options is not const 

### ISSUE THREAD

**Issue by smurthys** [sigcpp/stl-lite#29]
Parameter [`arguments` in function `get_options`](https://github.com/sigcpp/stl-lite/blob/87eef34616b5c5c377899f84f6bfb5b630b4501e/test/options.cpp#L28) is not `const`. For safety, that parameter should be marked `const`.

Making this change will also need a change in `main`, but an easy change.

**Comment by cmcmone** [comment]
I agree, change the parameter to const is better, to prevent the pointer from changing. Thanks @smurthys for catch this issue.

**Comment by smurthys** [comment]
@cmcmone We need to assure that `get_options` does not modify either the C-string pointers or the C-string content. I wrote a [SIGCPP blog post](https://sigcpp.github.io/2020/06/20/what-the-actual-const) about this topic:
